src/platform/windows/display_vram.cpp

```diff
@@ -910,97 +920,228 @@ namespace platf::dxgi {
+        }
+        img_out = *p_img;
+        break;
       }
 
-      device_ctx->OMSetBlendState(blend_disable.get(), nullptr, 0xFFFFFFFFu);
+      case ofa::copy_last_surface_and_blend_cursor: {
+        auto p_surface = std::get_if<texture2d_t>(&last_frame_variant);
+        if (!p_surface) {
+          BOOST_LOG(error) << "Logical error at " << __FILE__ << ":" << __LINE__;
+          return capture_e::error;
+        }
+        if (!draw_mouse_cursor_flag) {
+          BOOST_LOG(error) << "Logical error at " << __FILE__ << ":" << __LINE__;
+          return capture_e::error;
+        }
 
-      ID3D11RenderTargetView *emptyRenderTarget = nullptr;
-      device_ctx->OMSetRenderTargets(1, &emptyRenderTarget, nullptr);
-      device_ctx->RSSetViewports(0, nullptr);
-      ID3D11ShaderResourceView *emptyShaderResourceView = nullptr;
-      device_ctx->PSSetShaderResources(0, 1, &emptyShaderResourceView);
-    }
+        if (!pull_free_image_cb(img_out)) return capture_e::interrupted;
+
+        auto [d3d_img, lock] = get_locked_platform_img(img_out);
+        if (!d3d_img) return capture_e::error;
+
+        device_ctx->CopyResource(d3d_img->capture_texture.get(), p_surface->get());
 
-    // Release the mutex to allow encoding of this frame
-    img->capture_mutex->ReleaseSync(0);
+        {
+          device_ctx->VSSetShader(scene_vs.get(), nullptr, 0);
+          device_ctx->PSSetShader(scene_ps.get(), nullptr, 0);
+          device_ctx->OMSetRenderTargets(1, &d3d_img->capture_rt, nullptr);
+
+          if (cursor_alpha.texture.get()) {
+            // Perform an alpha blending operation
+            device_ctx->OMSetBlendState(blend_alpha.get(), nullptr, 0xFFFFFFFFu);
+
+            device_ctx->PSSetShaderResources(0, 1, &cursor_alpha.input_res);
+            device_ctx->RSSetViewports(1, &cursor_alpha.cursor_view);
+            device_ctx->Draw(3, 0);
+          }
+
+          if (cursor_xor.texture.get()) {
+            // Perform an invert blending without touching alpha values
+            device_ctx->OMSetBlendState(blend_invert.get(), nullptr, 0x00FFFFFFu);
+
+            device_ctx->PSSetShaderResources(0, 1, &cursor_xor.input_res);
+            device_ctx->RSSetViewports(1, &cursor_xor.cursor_view);
+            device_ctx->Draw(3, 0);
+          }
+
+          device_ctx->OMSetBlendState(blend_disable.get(), nullptr, 0xFFFFFFFFu);
+
+          ID3D11RenderTargetView *emptyRenderTarget = nullptr;
+          device_ctx->OMSetRenderTargets(1, &emptyRenderTarget, nullptr);
+          device_ctx->RSSetViewports(0, nullptr);
+          ID3D11ShaderResourceView *emptyShaderResourceView = nullptr;
+          device_ctx->PSSetShaderResources(0, 1, &emptyShaderResourceView);
+        }
+        break;
+      }
+
+      case ofa::dummy_fallback: {
+        if (!pull_free_image_cb(img_out)) return capture_e::interrupted;
+        auto img = (img_d3d_t *) img_out.get();
+        if (complete_img(img, true)) return capture_e::error;
```

Comment:
If this image was used before, it may have a cursor rendered into it. That's why the code was using `UpdateSubresource()` to clear it each time.

Reply:
Hmm, I think I would prefer to render the cursor. @psyke83 might have more comments about this, but I believe we do commonly hit this case where a game is loading and hasn't yet rendered its first frame. It might be confusing to the user to just see a black screen with nothing on it, especially because the mouse will be visible on the host display.

Reply:
> especially because the mouse will be visible on the host display

Makes sense, will do it.

Reply:
Done.